### PR TITLE
add peptide motif regex feature

### PR DESCRIPTION
**Pull request by Leon-Bichmann**
Added a flag to PSMFeatureExtractor to include Peptide Motifs as feature for Percolator.

This is similar to using the enzyme specificity feature for example from a tryptic digest in case you have special peptide motifs like in MHC binding peptides.

### CODE REVIEW

**Comment by timosachsenberg** [comment]
- please remove swap files
- can there be some bias introduced if the regex only matches targets (vs. decoys). If so this needs to be documented.
- please add at least a class test

**Comment by timosachsenberg** [comment]
Sounds like a useful addition. Thanks!

**Comment by hroest** [comment]
> not sure how the pyopenms bindings work.. ill have a look

see https://pyopenms.readthedocs.io/en/latest/wrap_classes.html# - basically you have to add the new function to the corresponding `pxd` file

File: src/tests/class_tests/openms/source/PercolatorFeatureSetHelper_test.cpp [hunk 1]
```diff
@@ -44,6 +45,20 @@
...
 
+bool check_regex(const OpenMS::String& peptide_motif)
+{
+    try
+    {
+      const OpenMS::String sequence = "TESTPEPTIDE";
+      boost::regex peptide_motif_regex(peptide_motif);
+      boost::regex_match(sequence, peptide_motif_regex);
+      return true;
+    } catch (boost::regex_error& e) {
```
Comment (hroest) [inline src/tests/class_tests/openms/source/PercolatorFeatureSetHelper_test.cpp]:
```suggestion
    } catch (boost::regex_error& e)
    {
```

File: src/openms/include/OpenMS/ANALYSIS/ID/PercolatorFeatureSetHelper.h [hunk 2]
```diff
@@ -96,7 +96,15 @@ namespace OpenMS
          * Concatenates SearchParameter of multiple search engine runs and merges PeptideEvidences, collects used search engines in MetaInfo for collection in addMULTISEFeatures for feature registration.
          */
         static void mergeMULTISEProteinIds(std::vector<ProteinIdentification>& all_protein_ids, std::vector<ProteinIdentification>& new_protein_ids);
-        
+
+        /**
+         * @brief addMotifRegExFeatures
```
Comment (hroest) [inline src/openms/include/OpenMS/ANALYSIS/ID/PercolatorFeatureSetHelper.h]: can you add a brief description hwere of what this does?

File: src/openms/source/ANALYSIS/ID/PercolatorFeatureSetHelper.cpp [hunk 3]
```diff
@@ -34,11 +34,43 @@
...
+        {
+          String sequence = hit->getSequence().toUnmodifiedString();
+
+          // set motif regex feature to 1 if hit sequence matches provided regex
+          if (boost::regex_match(sequence, peptide_motif_regex))
+          {
+
+            hit->setMetaValue("MotifRegEx:match", 1);
+
+          } else {
```
Comment (hroest) [inline src/openms/source/ANALYSIS/ID/PercolatorFeatureSetHelper.cpp]:
```suggestion
          } else
          {
```

**Pull request closed without merging**